### DISCUSSION TITLE
process_js_with_custom_pass API adjustment

### DISCUSSION THREAD

**Discussion by JSerFeng** [swc-project/swc#6287] [General]
Hello, last time I made a PR about making comments a parameter in api [`process_js_with_custom_pass()`](https://github.com/swc-project/swc/blob/main/crates/swc/src/lib.rs#L851), and I was wrong because my ignorance, you correct me that comments are stored in inner data, and can be cloned easily.

But now I still want to make `comments` outside of that API. Because it also need a Option<Program> as parameter, so If I'm right then I think I can give it an already parsed AST, but when I parse I also need a `comments` to parse, so the `comments` using for parsing and the `comments` for transforming should be the same.

I'm not sure though, I tried give a comments for parsing and a new one for `process_js_with_custom_pass`, and find out no comments in the result.

If I'm wrong again, sorry for bothering

**Comment by JSerFeng** [comment]
If you think this is fine, I'm very glad to PR

**Reply by kdy1** [reply]
I think you are right. It should accept comments in case the user is passing a `Program` instance
